profiling/src/profile/mod.rs

```diff
@@ -544,18 +558,40 @@ impl Profile {
     }
 
     /// Validates labels and converts them to the internal representation.
+    /// Extracts out the timestamp label, if it exists.
     /// Also tracks the index of the label with key "local root span id".
     fn extract_sample_labels(
         &mut self,
         sample: &api::Sample,
-    ) -> anyhow::Result<(Vec<Label>, Option<usize>)> {
+    ) -> anyhow::Result<(Vec<Label>, Option<usize>, Option<Timestamp>)> {
         let mut labels: Vec<Label> = Vec::with_capacity(sample.labels.len());
         let mut local_root_span_id_label_offset: Option<usize> = None;
+        let mut timestamp: Option<Timestamp> = None;
+
         for label in sample.labels.iter() {
             let key = self.intern(label.key);
             let str = label.str.map(|s| self.intern(s)).unwrap_or(0);
             let num_unit = label.num_unit.map(|s| self.intern(s)).unwrap_or(0);
 
+            if key == self.timestamp_key {
+                anyhow::ensure!(
+                    str == 0,
+                    "the label \"{}\" must be sent as a number, not string {}",
+                    label.str.unwrap(),
+                    self.get_string(self.timestamp_key).unwrap()
+                );
+                anyhow::ensure!(
+                    label.num != 0,
+                    "the label \"{}\" must not be 0",
+                    self.get_string(self.timestamp_key).unwrap()
+                );
+                //TODO what should I ensure the units are?
```

Comment:
I actually wonder if we should get rid of the units and take the memory savings that come with that -- I'm not sure any of our profilers ever sets the units.

Reply:
We never set any units on any label? Or only on timestamp labels?

Reply:
https://github.com/DataDog/libdatadog/issues/204 to track.  That feels like an orthogonal change that is best handled in a separate PR.

Reply:
I mean on any label. I did a quick check:

* [None in Ruby profiler](https://github.com/search?q=repo%3ADataDog%2Fdd-trace-rb+ddog_prof_Label&type=code)
* [Quick check in Python profiler seems none are used](https://github.com/DataDog/dd-trace-py/blob/2eef3c3cad7e95b49dc92a180e9bf4914b2380d5/ddtrace/internal/datadog/profiling/src/exporter.cpp#L438-L456)
* [Don't see it in dotnet](https://github.com/DataDog/dd-trace-dotnet/blob/25f329a4d1af371d1fe42ab6bbce56c734d768a2/profiler/src/ProfilerEngine/Datadog.Profiler.Native/LibddprofExporter.cpp#L383-L396)
* [Also not in the native profiler](https://github.com/DataDog/ddprof/blob/9c3b87f0da7a616f10733ad2d06038640f6fe8a8/src/pprof/ddprof_pprof.cc#L218-L265)

...ahaha but last and not least, [php does seem to be using it](https://github.com/DataDog/dd-trace-php/blob/1114d91b278e1aa3f8de0d06e31e4646744706ed/profiling/src/profiling/mod.rs#L86-L109). So I guess something to raise with our estemeed colleagues?